code/modules/mob/living/simple_animal/bot/secbot.dm

```diff
@@ -236,34 +249,33 @@
 	INVOKE_ASYNC(src, PROC_REF(cuff_callback), C)
 
 /mob/living/simple_animal/bot/secbot/proc/cuff_callback(mob/living/carbon/C)
-	if(do_after(src, 60, target = C))
-		if(!C.handcuffed && on)
-			C.handcuffed = new /obj/item/restraints/handcuffs/cable/zipties/used(C)
-			C.update_handcuffed()
-			playsound(loc, pick('sound/voice/bgod.ogg', 'sound/voice/biamthelaw.ogg', 'sound/voice/bsecureday.ogg', 'sound/voice/bradio.ogg', 'sound/voice/binsult.ogg', 'sound/voice/bcreep.ogg'), 50, 0)
-			back_to_idle()
+	currently_cuffing = TRUE
+	if(!do_after(src, 6 SECONDS, target = C))
+		currently_cuffing = FALSE
```

Comment:
if we're in the `do_after` guard, shouldn't we just return early, because we don't want currently_cuffing to be FALSE until after 6 seconds?

Reply:
if bot failed to cuff someone because it was moved, lets say pulled around by someone, it should instantly try to cuff again instead of waiting 6 seconds, why?
actually, currently if i just early return, bot will unable to cuff again until reset due to `currently_cuffing` being changed only in this proc and in `reset` proc

Reply:
the idea of that variable to stop bot from attempting cuffing again if its currently cuffing. If currently cuffing failed, it should be able to cuff